File: lib/android/src/main/java/com/rnadapty/react/models/AdaptyPaywall.kt
package com.rnadapty.react.models

import com.adapty.models.PaywallModel
import com.google.gson.annotations.SerializedName

class AdaptyPaywall(paywall: PaywallModel) {
    @SerializedName("revision")
    var revision = paywall.revision
    @SerializedName("visualPaywall")
    var visualPaywall = paywall.visualPaywall
    @SerializedName("products")
    var products: ArrayList<AdaptyProduct>? = ArrayList(paywall.products.map { AdaptyProduct(it) })
    @SerializedName("developerId")
    var developerId = paywall.developerId
    @SerializedName("isPromo")
    var isPromo = paywall.isPromo
    @SerializedName("variationId")
    var variationId = paywall.variationId
    @SerializedName("customPayloadString")
    var customPayloadString = paywall.customPayloadString
    @SerializedName("name")
    var name = paywall.name
    @SerializedName("abTestName")
    var abTestName = paywall.abTestName
}

File: lib/ios/Models/AdaptyPaywall.swift
//
//  AdaptyPaywall.swift
//  RNAdapty
//
//  Created by Ivan on 12.07.2021.
//

import Adapty
import Foundation

struct AdaptyPaywall: Encodable {
  let developerId: String
  let variationId: String
  let revision: Int
  let isPromo: Bool
  let products: [AdaptyProduct]
  let visualPaywall: String?
  let customPayloadString: String?
  let customPayload: String?
  let abTestName: String?
  let name: String?

  init(_ paywall: PaywallModel) {
    developerId = paywall.developerId
    variationId = paywall.variationId
    revision = paywall.revision
    isPromo = paywall.isPromo
    products = paywall.products.map { AdaptyProduct.init($0, paywall.variationId) }
    visualPaywall = paywall.visualPaywall
    customPayload = paywall.customPayloadString
    customPayloadString = paywall.customPayloadString
    abTestName = paywall.abTestName
    name = paywall.name
  }
}
